src/twisted/conch/unix.py

```diff
@@ -51,10 +52,12 @@
 class UnixSSHRealm:
     def requestAvatar(
         self,
-        username: bytes,
+        username: bytes | Tuple[()],
         mind: object,
         *interfaces: portal._InterfaceItself,
     ) -> Tuple[portal._InterfaceItself, UnixConchUser, Callable[[], None]]:
+        if not isinstance(username, bytes):
```

Comment:
I guess that this is fine, since this is the current behaviour.

----------

Just a comment on design.

I did take a look at twisted.cred a long time ago, and start using it, but then I switched to a custom implementation.

For anonymous I ended up implementing it using different credential type ( `IAnonymous` ) and then the realm doesn't have to care about anonymous.

The anonymous cred checker will reject or accept the anonymous user and return the avatar that will handle the anonymous access.

Not sure which is best.
I still think that the idea of pluggable auth method is nice... but things get more complicated when for example you implement OIDC + extra OAuth2 API access for Realm

Reply:
> I did take a look at twisted.cred a long time ago, and start using it, but then I switched to a custom implementation.

This is interesting for discussion but not really relevant to this ticket so I am resolving :)